modal/_utils/grpc_utils.py

```diff
@@ -174,8 +171,11 @@ async def unary_stream(
         yield item
 
 
+R = TypeVar("R")
+
+
 async def retry_transient_errors(
-    fn: "modal.client.UnaryUnaryWrapper[RequestType, ResponseType]",
+    fn: Callable[..., Awaitable[R]],
```

Comment:
I couldn't find a good way to get the typing to work with `UnaryUnaryWrapper` directly and the generics:

![Screenshot 2025-09-05 at 4 31 34 PM](https://github.com/user-attachments/assets/89d2293d-51b4-48c4-926a-14bd29686386)

So I went with passing a callable and then pass the name directly in `fn_name`.